core/src/drop_bank_service.rs

```diff
@@ -39,3 +40,204 @@ impl DropBankService {
         self.thread_hdl.join()
     }
 }
+
+#[cfg(test)]
+pub mod tests {
+    use {
+        super::*,
+        crate::{
+            consensus::{
+                heaviest_subtree_fork_choice::HeaviestSubtreeForkChoice,
+                progress_map::{ForkProgress, ProgressMap},
+            },
+            repair::cluster_slot_state_verifier::{
+                DuplicateConfirmedSlots, DuplicateSlotsTracker, EpochSlotsFrozenSlots,
+            },
+            replay_stage::ReplayStage,
+            unfrozen_gossip_verified_vote_hashes::UnfrozenGossipVerifiedVoteHashes,
+        },
+        crossbeam_channel::unbounded,
+        itertools::Itertools,
+        solana_ledger::genesis_utils::create_genesis_config,
+        solana_program_runtime::timings::ExecuteTimings,
+        solana_runtime::{
+            accounts_background_service::AbsRequestSender, bank::Bank, bank_forks::BankForks,
+            genesis_utils::GenesisConfigInfo, prioritization_fee_cache::PrioritizationFeeCache,
+        },
+        solana_sdk::{
+            hash::Hash,
+            pubkey::Pubkey,
+            system_transaction,
+            transaction::{Result, SanitizedTransaction},
+        },
+        solana_unified_scheduler_pool::{
+            DefaultTaskHandler, HandlerContext, PooledScheduler, SchedulerPool, TaskHandler,
+        },
+        std::{
+            collections::HashMap,
+            sync::{Arc, Mutex},
+        },
+    };
+
+    #[test]
+    fn test_scheduler_waited_by_drop_bank_service() {
```

Comment:
running this test without the fix reliably causes deadlock.

Reply:
sorry to nit-pick but this test really does not seem in the right spot to me.
it is not really testing the behavior of the DropBankService, it's testing the response of the scheduler, right?

Correct me if I'm wrong, but the fix for this was just taking ownership of the banks we want to drop for a bit longer, and then dropping them **outside** the write-lock of bank_forks, right?
So seems this test either belongs in scheduling (pool?) code, or maybe bank_forks?

Reply:
> Correct me if I'm wrong, but the fix for this was just taking ownership of the banks we want to drop for a bit longer, and then dropping them **outside** the write-lock of bank_forks, right?

this is all correct.

> So seems this test either belongs in scheduling (pool?) code, or maybe bank_forks?
Semantically, yes. But I actually wanted to use `DropBankService` in the test. so `solana-unified-scheduler` and `solana-runtime` aren't right crates... (i.e. `DropBankService` is defined at `solana-core`.

> it is not really testing the behavior of the DropBankService
Strictly speaking, it's testing the behavior of DropBankService, regarding dropping `DropWithSchdeduler`. However, as you said, it's not that important to who is dropping it as long as it's **outside** the write-lock of bank_forks.

That said, I think i've moved the test into more natural place: 2c5d444db417c2ad82089b7192f1b73bbb628958 (after all, this is kind of integration test)